core/modules/file/file.theme.inc

```diff
@@ -278,6 +279,14 @@ function theme_file_upload_help($variables) {
   if (isset($upload_validators['file_validate_extensions'])) {
     $descriptions[] = t('Allowed file types: !extensions.', array('!extensions' => '<strong>' . check_plain($upload_validators['file_validate_extensions'][0]) . '</strong>'));
   }
+  if (isset($cardinality)) {
+    if ($cardinality == -1) {
+      $descriptions[] = t('Unlimited number of files can be uploaded to this field.');
+    }
+    else {
+      $descriptions[] = format_plural($cardinality, 'This field can store only one file.', 'This field can store at most @count files.');
```

Comment:
When allowing a large but still specific amount of files (i.e. not unlimited), scrolling will be inevitable. In such cases, it would be great to have an indication of how many more can be uploaded; so perhaps change `@count` to `@max`, and then having something like `@count uploaded of @max allowed.`